src/Development/IDE/Core/Rules.hs

```diff
@@ -612,37 +612,27 @@ ghcSessionDepsDefinition file = do
 
 getModIfaceFromDiskRule :: Rules ()
 getModIfaceFromDiskRule = defineEarlyCutoff $ \GetModIfaceFromDisk f -> do
-  -- get all dependencies interface files, to check for freshness
-  (deps,_) <- use_ GetLocatedImports f
-  depHis  <- traverse (use GetModIface) (mapMaybe (fmap artifactFilePath . snd) deps)
-
   ms <- use_ GetModSummary f
-  let hiFile = toNormalizedFilePath'
-             $ case ms_hsc_src ms of
-                HsBootFile -> addBootSuffix (ml_hi_file $ ms_location ms)
-                _ -> ml_hi_file $ ms_location ms
-
-  case sequence depHis of
-    Nothing -> pure (Nothing, ([], Nothing))
-    Just deps -> do
-        mbHiVersion <- use  GetModificationTime hiFile
+  (diags_session, mb_session) <- ghcSessionDepsDefinition f
+  case mb_session of
+      Nothing -> return (Nothing, (diags_session, Nothing))
+      Just session -> do
+        let hiFile = toNormalizedFilePath'
+                    $ case ms_hsc_src ms of
+                        HsBootFile -> addBootSuffix (ml_hi_file $ ms_location ms)
+                        _ -> ml_hi_file $ ms_location ms
+        mbHiVersion <- use  (GetModificationTime_ False) hiFile
```

Comment:
Looking at the use site, a newtype or a custom ADT instead f a boolean might be nice.

Reply:
A newtype will be too verbose, given that there is already a field name, and a custom ADT will incurr in lots of instances for Shake.

I prefer to change the use site to show the field name.